src/build/app/mod.rs

```diff
@@ -2292,47 +2277,50 @@ impl<'help> App<'help> {
     #[doc(hidden)]
     pub fn _build(&mut self) {
         debug!("App::_build");
+        if !self.settings.is_set(AppSettings::Built) {
+            // Make sure all the globally set flags apply to us as well
+            self.settings = self.settings | self.g_settings;
 
-        // Make sure all the globally set flags apply to us as well
-        self.settings = self.settings | self.g_settings;
-
-        self._propagate(Propagation::Full);
```

Comment:
@pksunkara You can check this, the old code recursivly propagates to all it's subcommands, no matter it's used or not.